qiskit/aqua/algorithms/vq_algorithm.py

```diff
@@ -59,27 +59,42 @@ def __init__(self,
             initial_point: An optional initial point (i.e. initial parameter values)
                 for the optimizer.
             quantum_instance: Quantum Instance or Backend
+
         Raises:
              ValueError: for invalid input
         """
         super().__init__(quantum_instance)
 
-        self._var_form = var_form
         self._optimizer = optimizer
         self._cost_fn = cost_fn
         self._initial_point = initial_point
+        self.var_form = var_form
 
         self._parameterized_circuits = None
 
     @property
-    def var_form(self) -> Optional[VariationalForm]:
+    def var_form(self) -> Optional[Union[QuantumCircuit, VariationalForm]]:
         """ Returns variational form """
         return self._var_form
 
     @var_form.setter
-    def var_form(self, var_form: VariationalForm):
+    def var_form(self, var_form: Union[QuantumCircuit, VariationalForm]):
         """ Sets variational form """
-        self._var_form = var_form
+        if isinstance(var_form, QuantumCircuit):
+            # store the parameters
+            self._var_form_params = list(var_form.parameters)
+            self._var_form = var_form
+        elif isinstance(var_form, VariationalForm):
+            self._var_form_params = ParameterVector('θ', length=var_form.num_parameters)
+            self._var_form = var_form
+        elif var_form is None:
```

Comment:
Same comment about None as feature map i.e. should just be a valid value.

Reply:
Changed 👍